### PR TITLE
Add a pruning example with PyTorch Ignite integration module.

### PR DESCRIPTION
**Pull request by toshihikoyanase**
This PR adds a new example to show the usage of [`optuna.integration.IgnitePruningHandler`](https://optuna.readthedocs.io/en/latest/reference/integration.html#optuna.integration.IgnitePruningHandler).

Similarly to [the example of early stopping](https://github.com/pytorch/ignite/blob/master/examples/notebooks/TextCNN.ipynb) in the ignite repository, it uses two evaluators. One is for training data, and the other one is for validation data.

If we use the same evaluator for both training data and validation data, both training accuracy and validation accuracy will be reported as intermediate values. In such a situation, the pruning will not work correctly.

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
This PR was taken over https://github.com/pfnet/optuna/pull/633 because pruning examples will be integrated into plain ones.
